src/modules/pos/validator/components/EditStake/index.js

```diff
@@ -97,6 +107,26 @@ const EditStake = ({ history, stakeEdited, network, staking }) => {
   const mode = validatorStake || staking[address] ? 'edit' : 'add';
   const titles = getTitles(t)[mode];
 
+  const isMonthly = selectedDuration === REWARD_DURATIONS.monthly;
+  const queryConfig = {
+    options: {
+      enabled: !!validator.address && !stakeAmount.error && Number(stakeAmount.value) > 0,
+    },
+    config: {
+      params: {
+        validatorAddress: validator.address,
+        stake: convertToBaseDenom(stakeAmount.value, token),
+        validatorReward: '0',
+      },
+    },
+  };
+
+  const debouncedQueryConfig = useDebounce(queryConfig, 500);
```

Comment:
this looks strange 🤔  Why this debounce?

Reply:
because the users can enter stake amount and the API will be triggered for every input changes.. which is unnecessary API calls for estimations?

Reply:
true, looks good when looking at it more closely 👍